src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotPrimaryDataQueryInterface.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.grammar.Query;
+import bio.terra.grammar.exception.InvalidQueryException;
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.model.SnapshotRequestQueryModel;
+import bio.terra.service.common.CommonFlightUtils;
+import bio.terra.service.dataset.AssetSpecification;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.snapshot.exception.AssetNotFoundException;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import java.time.Instant;
+import java.util.List;
+import org.springframework.stereotype.Component;
+
+@Component
```

Comment:
interesting...I'm surprised that you needed the `@Component` annotation

Reply:
Oh, good catch, I don't think I need it. Removed.